sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/WebApplicationFirewallPolicy.java

```diff
@@ -0,0 +1,533 @@
+
+    /** @return Web Application Firewall managed rules */
+    ManagedRulesDefinition getManagedRules();
+
+    /** @return whether this policy is enabled */
+    boolean isEnabled();
+
+    /** @return an immutable list of application gateway resource ids associated with this Web Application Firewall Policy */
+    List<String> getAssociatedApplicationGatewayIds();
+
+    /**
+     * Get an immutable list of application gateways associated with this Web Application Firewall Policy by calling REST API.
+     *
+     * @return an immutable list of application gateways associated with this Web Application Firewall Policy
+     */
+    List<ApplicationGateway> getAssociatedApplicationGateways();
+
+    /**
+     * Get a list of application gateways associated with this Web Application Firewall Policy by calling REST API
+     * in async manner.
+     *
+     * @return {@link Flux} of application gateways associated with this Web Application Firewall Policy
+     */
+    Flux<ApplicationGateway> getAssociatedApplicationGatewaysAsync();
+
+    /**
+     * The entirety of the Web Application Firewall Policy definition.
+     */
+    interface Definition
+        extends DefinitionStages.Blank,
+        DefinitionStages.WithGroup,
+        DefinitionStages.WithCreate,
+        DefinitionStages.WithRequestBodyOrCreate,
+        DefinitionStages.WithManagedRulesOrCreate {
+    }
+
+    /** Grouping of Web Application Gateway stages. */
+    interface DefinitionStages {
+        /** The first stage of an application gateway definition. */
+        interface Blank extends DefinitionWithRegion<WithGroup> {
+        }
+
+        /** The stage of a Web Application Firewall Policy definition allowing to specify the resource group. */
+        interface WithGroup extends GroupableResource.DefinitionStages.WithGroup<WithManagedRules>{
+        }
+
+        /** The stage of a Web Application Firewall Policy definition allowing to specify the managed rules. */
+        interface WithManagedRules {
+            /**
+             * Specifies a known managed rule set to be added to the Web Application Firewall.
+             *
+             * @param managedRuleSet known managed rule set
+             * @return the next stage of the definition
+             */
+            WithManagedRulesOrCreate withManagedRuleSet(KnownManagedRuleSet managedRuleSet);
+
+            /**
+             * Specifies a known managed rule set to be added to the Web Application Firewall, with a rule group override
+             * configuration.
+             *
+             * @param managedRuleSet known managed rule set
+             * @param managedRuleGroupOverrides rule group override configuration
+             * @return the next stage of the definition
+             */
+            WithManagedRulesOrCreate withManagedRuleSet(KnownManagedRuleSet managedRuleSet,
+                                                        ManagedRuleGroupOverride... managedRuleGroupOverrides);
```

Comment:
Does the vargs handle the former `WithManagedRulesOrCreate withManagedRuleSet(KnownManagedRuleSet managedRuleSet);`?

Reply:
Good catch...